### PR TITLE
Protected fields

### PR DESCRIPTION
**Pull request by Pikkupomo**
Add better error message when trying to save shop that has protection triggered.
Add protection handling to `MultiLanguageModelForm`

Refs SHOOP-1896

### CODE REVIEW

File: shoop/utils/multilanguage_model_form.py [hunk 1]
```diff
@@ -87,6 +87,12 @@ def __init__(self, **kwargs):
         kwargs["initial"] = object_data
         super(MultiLanguageModelForm, self).__init__(**kwargs)
 
+        if all([self.instance, self.instance.pk, hasattr(self.instance, "protected_fields")]):
```
Comment (akx) [inline shoop/utils/multilanguage_model_form.py]: I'm not sure this is very good use of `all` :) Also, that oughta be `getattr` with an empty tuple as the default arg to be clear...

Comment (akx) [inline shoop/utils/multilanguage_model_form.py]: You're using this same idiom (`if ...: for field in ...:`) in both these functions; might be worth to refactor it to

```python
def _get_protected_fields(self):
    if self.instance and self.instance.pk:
        return getattr(self.instance, "protected_fields", ())
    return ()
```

(with a suitable docstring, of course ;) )

File: shoop/admin/modules/shops/views/edit.py [hunk 2]
```diff
@@ -6,19 +6,20 @@
 # This source code is licensed under the AGPLv3 license found in the
 # LICENSE file in the root directory of this source tree.
 from __future__ import unicode_literals
-
 from django.conf import settings
 from django.utils.translation import ugettext_lazy as _
-
 from shoop.admin.forms.widgets import MediaChoiceWidget
```
Comment (suutari-ai) [inline shoop/admin/modules/shops/views/edit.py]: Why are you removing the grouping of imports?

Comment (akx) [inline shoop/admin/modules/shops/views/edit.py]: Probably PyCharm being overzealous.

**Pull request merged**